tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2577,6 +2566,197 @@ llvm::Optional<Value> convertReduceMeanOp(
+        return llvm::None;
+      }
+
+      RankedTensorType output_bool_type =
+          RankedTensorType::get(output_type.getShape(), rewriter.getI1Type());
+
+      auto resize_op = rewriter.create<tosa::ResizeOp>(
+          op->getLoc(), output_acc_type, input_value, output_size, stride,
+          offset, shift_attr, rewriter.getF32ArrayAttr({0.0, 0.0}),
+          rewriter.getF32ArrayAttr({0.0, 0.0}), resize_mode);
+
+      // TFLite resize_bilinear always assume input and output tensors have same
+      // scale That means we only need to arithmetic right shift with (2 *
+      // shift)
+      // TODO: Align TFLite rounding behavior
+      // TFLite also uses symmetric rounding by doing 'x / (1 << 20)'
+      // TOSA arithmetic right shift is doing standard rounding.
+      // Right now it's legalized using GreaterEqualOp + SelectOp to conform to
+      // TFLite reference. But this eventually should be fixed in TFLite
+      // reference
+      Value cst_zero = getTosaConstTensorSingleI32(rewriter, op, 0);
+      Value cst_twenty = getTosaConstTensorSingleI32(rewriter, op, 20);
+
+      auto ge_op = rewriter.create<tosa::GreaterEqualOp>(
+          op->getLoc(), output_bool_type, resize_op.getResult(), cst_zero);
+
+      auto abs_op = rewriter.create<tosa::AbsOp>(op->getLoc(), output_acc_type,
+                                                 resize_op.getResult());
+
+      auto rshift_op = rewriter.create<tosa::ArithmeticRightShiftOp>(
+          op->getLoc(), output_acc_type, abs_op.getResult(), cst_twenty, true);
+
+      auto negate_op = rewriter.create<tosa::NegateOp>(
+          op->getLoc(), output_acc_type, rshift_op.getResult());
+
+      auto select_op = rewriter.create<tosa::SelectOp>(
+          op->getLoc(), output_acc_type, ge_op.getResult(),
+          rshift_op.getResult(), negate_op.getResult());
+
+      auto cast_op = rewriter.create<tosa::CastOp>(op->getLoc(), output_type,
+                                                   select_op.getResult());
+
+      val = cast_op.getResult();
+    } else if (mode == "NEAREST_NEIGHBOR") {
+      auto resize_op = rewriter.create<tosa::ResizeOp>(
+          op->getLoc(), output_type, input_value, output_size, stride, offset,
+          shift_attr, rewriter.getF32ArrayAttr({0.0, 0.0}),
+          rewriter.getF32ArrayAttr({0.0, 0.0}), resize_mode);
+      val = resize_op.getResult();
+    } else {
+      op->emitOpError(
+          "OpResize: only support BILINEAR or NEAREST_NEIGHBOR mode");
+      return llvm::None;
+    }
+  } else {
+    auto resize_op = rewriter.create<tosa::ResizeOp>(
+        op->getLoc(), output_type, input_value, output_size,
+        rewriter.getI64ArrayAttr({0, 0}), rewriter.getI64ArrayAttr({0, 0}),
+        rewriter.getI32IntegerAttr(0),
+        rewriter.getF32ArrayAttr({fp_stride_y, fp_stride_x}),
+        rewriter.getF32ArrayAttr({fp_offset_y, fp_offset_x}), resize_mode);
+    val = resize_op.getResult();
+  }
+  return val;
```

Comment:
nit: prefer to return in each branch of the condition above vs joining and then returning an intermediate.

Reply:
Thanks. Will update in next round.